service/src/main/java/bio/terra/externalcreds/dataAccess/SshKeyPairDAO.java

```diff
@@ -0,0 +1,81 @@
+package bio.terra.externalcreds.dataAccess;
+
+import bio.terra.externalcreds.generated.model.SshKeyPairType;
+import bio.terra.externalcreds.models.SshKeyPair;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Optional;
+import org.springframework.dao.support.DataAccessUtils;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class SshKeyPairDAO {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  public SshKeyPairDAO(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public Optional<SshKeyPair> getSshKeyPair(String userId, SshKeyPairType type) {
+    var namedParameters =
+        new MapSqlParameterSource().addValue("userId", userId).addValue("type", type.name());
+    var query = "SELECT * FROM ssh_key_pair WHERE user_id = :userId AND type = :type";
```

Comment:
As a preference, I avoid using the `SELECT *` form in favor of an explicit list. In this case it is not a risk since the row mapper is in charge of the field extraction.

Reply:
done